### PR TITLE
Restyle Gesture Handler Documentation

### PR DESCRIPTION
**Pull request by patrycjakalinska**
New branch with fixed Docusaurus - changes include that done [prior PR.](https://github.com/software-mansion/react-native-gesture-handler/pull/2737)

Additionally:
* Add `Playground` with examples: Pan, Tap, Rotation, Fling, LongPress and Pinch
* Add dark mode
* Make landing page responsive and well-adjusted for mobile, tablet, laptop and laptop+ dimensions
* Add reusable components: `HomepageButton` and `InteractiveExampleComponent`

### CODE REVIEW

File: docs/src/components/GestureExamples/TapExample/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      pressed.value = false;
+    });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    backgroundColor: pressed.value
+      ? 'var(--swm-yellow-dark-80)'
+      : 'var(--swm-purple-light-100)',
+    // border: pressed.value
+    //   ? 'var(--swm-red-dark-80)'
+    //   : '8px solid var(--swm-purple-light-80)',
```
Comment (kacperkapusciak) [inline docs/src/components/GestureExamples/TapExample/index.tsx:33]:
```suggestion
```
remove dead code

File: docs/src/components/GestureFeatures/GestureFeatureList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
+  },
+  {
+    title: 'DECLARATIVE',
+    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
+  },
+  {
+    title: 'FEATURE',
+    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
+  },
```
Comment (kacperkapusciak) [inline docs/src/components/GestureFeatures/GestureFeatureList/index.tsx:17]:
```suggestion
  {
    title: 'native gesture recognizers',
    body: 'With Gesture Handler touch stream handling happens on the UI thread and uses APIs native to each platform.',
  },
  {
    title: 'native components',
    body: 'Gesture Handler library ships with a set of components that aims to provide best possible interations such as SwipeableRow or Drawer.',
  },
  {
    title: '120 FPS',
    body: 'Gesture Handler integrates tightly with Reanimated to allow you to build smooth gesture based experiences up to 120 fps.',
  },
```

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]

File: docs/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "react-native-gesture-handler-docs",
+  "name": "react-native-reanimated-docs",
   "version": "0.0.0",
-  "private": true,
+  "private": false,
```
[the remark this answers is not in the document]
Comment (patrycjakalinska) [reply docs/package.json:4]: It was changed to test something on mobile, my bad

File: docs/src/components/GestureExamples/FlingExample/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      startPositionX.value = e.x;
+      startPositionY.value = e.y;
+    })
+    .onEnd((e) => {
+      endPositionX.value = e.x;
+      endPositionY.value = e.y;
+    })
+    .onStart((e) => {
+      if (endPositionX.value === 0) endPositionX.value = e.x;
+      if (endPositionY.value === 0) endPositionY.value = e.y;
```
Comment (kacperkapusciak) [inline docs/src/components/GestureExamples/FlingExample/index.tsx:46]:
```suggestion
    .onStart((e) => {
      endPositionX.value = e.x;
      endPositionY.value = e.y;
```
please make this change to make it work a little bit more smooth

File: docs/src/components/GestureExamples/PanExample/index.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      offsetY.value =
+        Math.abs(offsetY.value) <= 100
+          ? offsetY.value + event.changeY >= 100
+            ? 100
+            : offsetY.value + event.changeY <= -100
+            ? -100
+            : offsetY.value + event.changeY
+          : offsetY.value;
+    })
+    .onEnd(() => setShowHand(true));
```
Comment (kacperkapusciak) [inline docs/src/components/GestureExamples/PanExample/index.tsx:43]:
```suggestion
    .onEnd(() => setShowHand(true))
    .minDistance(0);
```
add this here to make it more responsive

File: docs/src/components/GestureExamples/TapExample/index.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    })
+    .onFinalize(() => {
+      pressed.value = false;
+    });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    backgroundColor: pressed.value
+      ? 'var(--swm-yellow-dark-80)'
+      : 'var(--swm-purple-light-100)',
+    transform: [{ scale: withTiming(pressed.value ? 1.2 : 1) }],
```
Comment (kacperkapusciak) [inline docs/src/components/GestureExamples/TapExample/index.tsx:38]:
```suggestion
    transform: [{ scale: withTiming(pressed.value ? 1.8 : 1) }],
```
lets make it a bit bigger coz I have big fingers that fully cover the button :)

File: docs/src/components/GestureExamples/LongPressExample/index.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    .onEnd(() => {
+      pressed.value = false;
+      setShowHand(true);
+    });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    backgroundColor: pressed.value
+      ? 'var(--swm-yellow-dark-80)'
+      : 'var(--swm-purple-light-100)',
+    transform: [{ scale: withTiming(pressed.value ? 1.2 : 1) }],
```
Comment (kacperkapusciak) [inline docs/src/components/GestureExamples/LongPressExample/index.tsx:36]:
```suggestion
    transform: [{ scale: withTiming(pressed.value ? 1.8 : 1) }],
```
lets make it a bit bigger coz I have big fingers that fully cover the button :)

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]
image


it looks like the images of hands have incorrect fingers for most of the gestures
- pan should have 1 finger
- tap should have 1 finger
- rotate should either have 2 fingers, the arrow is perfect
- fling is ok
- longpress should have 1 finger
- pinch is perfect

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]
the version switch is missing so there's no way to change the docs version to v1 in the UI
Screenshot 2024-02-14 at 15 35 33

**Comment by kacperkapusciak** [comment]
I also don't like the fact the theme switcher is in an arbitrary spot on the landing page.
https://github.com/software-mansion/react-native-gesture-handler/assets/39658211/7c319298-605a-4260-b492-23f8763cbf45

Can we move it to the right hand side the same it is in the docs? 🙏

File: docs/src/components/GesturePlayground/index.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  },
+];
+
+const GesturePlayground = () => {
+  return (
+    <div>
+      <div>
+        <h2 className={styles.heading}>Learn how it works</h2>
+        <p className={styles.subheading}>
+          Hover over cards to see our interaction.
```
Comment (kacperkapusciak) [inline docs/src/components/GesturePlayground/index.tsx:44]:
```suggestion
          Tap and drag the circles to explore the gestures.
```

**Review by kacperkapusciak — APPROVED** [review]
LGTM :shipit:

**Pull request merged**